src/badguy/granito.hpp

```diff
@@ -45,37 +58,127 @@ class Granito : public WalkingBadguy
 
   virtual void kill_fall() override;
 
+  virtual ObjectSettings get_settings() override;
   virtual GameObjectTypes get_types() const override;
   virtual void after_editor_set() override;
 
+  virtual GranitoBig* get_carrier() const;
+  void turn(const Direction& direction);
+
+  /**
+   * @scripting
+   * @description Makes the Granito wave.
+   */
+  virtual void wave();
+
+  /**
+   * @scripting
+   * @description Makes the Granito sit.
+   */
+  virtual void sit();
+
+  /**
+   * @scripting
+   * @description Makes the Granito sit.
+   * @param string $direction Direction to turn to. Can be "left or "right".
+   */
+  void turn(const std::string& direction);
+
+  /**
+   * @scripting
+   * @description Sets the walking state for the Granito.
+   * @param bool $walking
+   */
+  void set_walking(bool walking)
+  {
+    if (walking)
+      walk();
+    else
+      stand();
+  }
+
+  /**
+   * @scripting
+   * @description Makes the Granito walk.
+   */
+  void walk();
+
+  /**
+   * @scripting
+   * @description Makes the Granito walk for a specified amount of seconds.
+   * @param float $seconds
+   */
+  void walk_for(float seconds);
+
+  /**
+   * @scripting
+   * @description Makes the Granito stand, or stop if walking.
+   */
+  void stand();
+
+  /**
+   * @scripting
+   * @description Makes the Granito jump.
+   */
+  virtual void jump();
+
+  /**
+   * @scripting
+   * @description Eject itself from the Big Granito.
+   */
+  virtual void eject();
+
+  /**
+   * @scripting
+   * @description Gets the current Granito state. Value is any of the ""GRANITO_STATE"" enumerators.
+   */
+  int get_state() const { return static_cast<int>(m_state); }
+
+  /**
+   * @scripting
+   * @description Gets the name of the Big Granito that is carrying the Granito.
+   */
+  std::string get_carrier_name() const;
+
+  /**
+   * @scripting
+   * @description Resets the detection checker,
```

Comment:
all hail the detection checker. all praise the detection checker. all admire the detection checker. the best invention made by mankind.

Reply:
huh? What kind of review comment is this?

Reply:
Me reminding myself that i need to change that description because i accidentally wrote "the detection checker" which sounds extremely wrong